tests/pkg_yxml/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ * @file
+ * @brief       Tests for the yxml XML parser library package
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+#include <string.h>
+#include <stdio.h>
+
+#include "embUnit.h"
+#include "yxml.h"
+
+static char _yxml_buf[1024];
+
+static const char *_doc = "<foo>bar</foo>";
+
+static void test_yxml(void)
+{
+    yxml_t yxml;
+    char content[64];
+
+    /* initialize yxml instance */
+    yxml_init(&yxml, _yxml_buf, sizeof(_yxml_buf));
+
+    /* parse XML document bytewise */
+    for (const char *doc = _doc; *doc; doc++) {
+        yxml_ret_t r = yxml_parse(&yxml, *doc);
+        TEST_ASSERT(r >= 0);
+
+        /* handle tokens */
+        switch (r) {
+            case YXML_OK:
+                break;
+            case YXML_ELEMSTART:
+                printf("YXML_ELEMSTART \"%s\"\n", yxml.elem);
```

Comment:
Are those `printf`s really needed? They mostly just destroy the structured output of `embunit`, which could be destructive, if we ever want to render its output XML (when compiled with `OUTPUT=XML`).

Reply:
uff. the test script *could* be extended to actually require this output. The test itself is using embUnit as a harness.
How about we remove the output when we do `OUTPUT=XML`? Can also do it now, I don't care much.

Reply:
Then why use `embunit` at all if you don't want to use its features?

Reply:
Because as is, it can be extended. Like, actually check if the right tokens arrive at the right time. But for this basic test, I just wanted to link in the library and make it parse something. 

Also, some features are used, like printing the `OK (n tests)` and the corresponding python test script.

Reply:
Then let's rather use `DEBUG()` here in case there is some checking to be done.

Reply:
discussed offline, changed to use DEBUG et al.